python/equistore-operations/equistore/operations/allclose.py

```diff
@@ -1,10 +1,81 @@
-import numpy as np
-
 from equistore.core import TensorBlock, TensorMap
 
 from . import _dispatch
-from ._utils import _check_blocks, _check_same_gradients, _check_same_keys
-from .equal import NotEqualError
+from ._utils import (
+    NotEqualError,
+    _check_blocks_impl,
+    _check_same_gradients_impl,
+    _check_same_keys_impl,
+)
+
+
+def _allclose_impl(
+    tensor_1: TensorMap, tensor_2: TensorMap, rtol: float, atol: float, equal_nan: bool
+) -> str:
+    """Abstract function to perform an allclose operation between two TensorMaps."""
+    message = _check_same_keys_impl(tensor_1, tensor_2, "allclose")
+    if message != "":
+        return f"the tensor maps have different keys: {message}"
+
+    for key, block_1 in tensor_1.items():
+        message = _allclose_block_impl(
+            block_1=block_1,
+            block_2=tensor_2.block(key),
+            rtol=rtol,
+            atol=atol,
+            equal_nan=equal_nan,
+        )
+        if message != "":
+            return f"blocks for key '{key}' are different: {message}"
```

Comment:
```suggestion
            return f"blocks for {key.print()} are different: {message}"
```

This will produce `blocks for (aa=3, bb=4) are different` instead of `blocks for key LabelsEntry(aa=3, bb=4) are different`

Reply:
(we can not use different `__repr__` and `__str__` for this, since torch did not support customizing `__repr__`, and this will only be available in torch >2.1)